internal/service/federatedsettingsidentityprovider/model_federated_settings_identity_provider.go

```diff
@@ -4,12 +4,73 @@ import (
 	"strings"
 
-	admin20231115008 "go.mongodb.org/atlas-sdk/v20231115008/admin"
+	"go.mongodb.org/atlas-sdk/v20231115014/admin"
 
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
 	"github.com/mongodb/terraform-provider-mongodbatlas/internal/common/conversion"
+	"github.com/spf13/cast"
 )
 
-func FlattenFederatedSettingsIdentityProvider(federatedSettingsIdentityProvider []admin20231115008.FederationIdentityProvider) []map[string]any {
+const WORKFORCE = "WORKFORCE"
+
+func ExpandIdentityProviderOIDCCreate(d *schema.ResourceData) *admin.FederationOidcIdentityProviderUpdate {
+	return &admin.FederationOidcIdentityProviderUpdate{
+		Audience:          conversion.StringPtr(d.Get("audience").(string)),
+		AssociatedDomains: expandAssociatedDomains(d),
+		AuthorizationType: conversion.StringPtr(d.Get("authorization_type").(string)),
+		ClientId:          conversion.StringPtr(d.Get("client_id").(string)),
+		Description:       conversion.StringPtr(d.Get("description").(string)),
+		DisplayName:       conversion.StringPtr(d.Get("name").(string)),
+		GroupsClaim:       conversion.StringPtr(d.Get("groups_claim").(string)),
+		IdpType:           conversion.StringPtr(WORKFORCE),
+		IssuerUri:         conversion.StringPtr(d.Get("issuer_uri").(string)),
+		Protocol:          conversion.StringPtr(d.Get("protocol").(string)),
+		RequestedScopes:   expandRequestedScopes(d),
+		UserClaim:         conversion.StringPtr(d.Get("user_claim").(string)),
+	}
+}
+func expandRequestedScopes(d *schema.ResourceData) *[]string {
+	requestedScopes := d.Get("requested_scopes")
+	requestedScopesSlice := cast.ToStringSlice(requestedScopes)
+	if requestedScopesSlice == nil {
+		requestedScopesSlice = []string{}
+	}
+	return &requestedScopesSlice
+}
+
+func expandAssociatedDomains(d *schema.ResourceData) *[]string {
+	associatedDomains := d.Get("associated_domains")
+	associatedDomainsSlice := cast.ToStringSlice(associatedDomains)
+	if associatedDomainsSlice == nil {
+		associatedDomainsSlice = []string{}
+	}
+	return &associatedDomainsSlice
+}
+
+func ExpandIdentityProviderUpdate(d *schema.ResourceData, existingIdentityProvider *admin.FederationIdentityProvider) *admin.FederationIdentityProviderUpdate {
+	return &admin.FederationIdentityProviderUpdate{
+		AssociatedDomains:          existingIdentityProvider.AssociatedDomains,
+		Audience:                   existingIdentityProvider.Audience,
+		AuthorizationType:          existingIdentityProvider.AuthorizationType,
+		ClientId:                   existingIdentityProvider.ClientId,
+		Description:                existingIdentityProvider.Description,
+		DisplayName:                existingIdentityProvider.DisplayName,
+		GroupsClaim:                existingIdentityProvider.GroupsClaim,
+		IdpType:                    existingIdentityProvider.IdpType,
+		IssuerUri:                  existingIdentityProvider.IssuerUri,
+		Protocol:                   existingIdentityProvider.Protocol,
+		PemFileInfo:                nil,
```

Comment:
Is this nil value correct?

Reply:
We left it as it was [before](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/2310/files#diff-6a7f7172746713f8b0bae40618f06b2903eec6958d6b1ff8f1c9a562d2ca10f9L316) because this is related to SAML and we didn't want to modify anything related to SAML if possible. From a quick look it seems that this attribute exists in the API but in terrafrom is only returned in the data sources. We want to investigate this further and depending on the time it takes will fix it in this project or create a ticket.